packages/order/src/utils/actions/receive-damaged-return-item.ts

```diff
@@ -0,0 +1,88 @@
+import { MedusaError, isDefined } from "@medusajs/utils"
+import { EVENT_STATUS } from "@types"
+import { ChangeActionType } from "../action-key"
+import { OrderChangeProcessing } from "../calculate-order-change"
+
+OrderChangeProcessing.registerActionType(
+  ChangeActionType.RECEIVE_DAMAGED_RETURN_ITEM,
+  {
+    isDeduction: true,
+    commitsAction: "return_item",
+    operation({ action, currentOrder, previousEvents }) {
+      const existing = currentOrder.items.find(
+        (item) => item.id === action.details.reference_id
+      )!
+
+      let toReturn = action.details.quantity
+
+      existing.return_dismissed_quantity ??= 0
+      existing.return_dismissed_quantity += toReturn
+      existing.return_requested_quantity -= toReturn
+
+      if (previousEvents) {
+        for (const previousEvent of previousEvents) {
+          previousEvent.original_ = JSON.parse(JSON.stringify(previousEvent))
+
+          let ret = Math.min(toReturn, previousEvent.details.quantity)
+          toReturn -= ret
+
+          previousEvent.details.quantity -= ret
+          if (previousEvent.details.quantity <= 0) {
+            previousEvent.status = EVENT_STATUS.DONE
+          }
+        }
+      }
+
+      return existing.unit_price * action.details.quantity
```

Comment:
note: these calculations will have to be done with some sort of `unit_total` that takes discounts and taxes into account as well. Doesn't have to go into this PR but worth keeping in mind.

Reply:
Yeah, here is not the only place where this will happen. I believe we'll have a util shared with multiple modules that can calculate theses things.
I will revisit that anyway when calculating using BigNumbers

Reply:
Yep sounds good